module-system/sov-modules-api/src/lib.rs

```diff
@@ -340,6 +340,23 @@ pub trait Module {
     }
 }
 
+impl Event for () {
+    fn event_key(&self) -> &'static str {
+        unimplemented!("Event type for this module does not exist.")
+    }
+}
```

Comment:
I know I told you to go ahead and do this, but I'm having second thought now 😅. 

With the current setup, there might a possibility that someone who meant to have no events could accidentally leave an `event.emit()` in their module and cause a `panic` if a user ever triggered that code path. Two potential options are:
1. Switch to a `NonInstantiable` event type which has no constructor - so the compiler will prevent them from ever emitting the event
2. Change this to emit a dummy key like `"()"` or `"nil"`

Reply:
I think it makes sense to prevent user from emitting the event, so I like switching to `NonInstantiable`. 

Can you explain how this prevents that though?

Reply:
TBF, the downside of `NonInstantiable` is that a module that does not have call messages are usually imlepmented as type CallMessage = (), so having type Event = NonInstantiable is a bit surprising for the user.

Reply:
Yeah. The reason for this is that `Callmessage`s have to be serializable and deserializable - which by gives users a way to instantiate them and force the rollup to deal with it.

Reply:
We used to have `CallMessage = NonInstantiable`, but it was insecure for that reason. If a user made a transaction for (say) BlobStorage (which had a `NonInstantiable` CallMessage, the rollup would be forced to process it. That processing *should* just be a no-op - but because the type claimed to be non-instantiable devs would sometimes assume they could panic instead

Reply:
Switched to emitting dummy key with warning after our offline discussion.